### PR TITLE
Code style tools

### PR DESCRIPTION
**Pull request by theunknownxy**
This implements the stuff discussed at #71.

- [x] editorconfig
- [x] clang-format(This needs a bit more time, because I'll try to match the existing codebase.)

### CODE REVIEW

**Comment by theunknownxy** [comment]
I added a config for clang-format. Feel free to try it out on some source files and tell me whether you like it. If you don't like it or think it doesn't fit the current style just let me know.

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+TabWidth: 4
+ContinuationIndentWidth: 2
+UseTab: ForIndentation
+ColumnLimit: 100
+
+AccessModifierOffset: -4
+NamespaceIndentation: All
+IndentCaseLabels: true
+
+SpaceBeforeParens: Never
```
Comment (yvt) [inline .clang-format]: Can you change `SpaceBeforeParens` to `ControlStatements`?

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+PointerAlignment: Left
```
Comment (yvt) [inline .clang-format]: Remove `PointerAlignment` as left aligned pointers are error-prone (e.g., `int* a, b`)

**Comment by theunknownxy** [comment]
@yvt I changed the lines according to your comments.

**Comment by yvt** [comment]
Thanks!

**Pull request merged**